pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/DefaultCommitterRealtimeIntegrationTest.java

```diff
@@ -0,0 +1,227 @@
+  private final long END_OFFSET = 500L;
+  private File TEMP_FILE;
+  private static final String SEGMENT_NAME = "testTable_126164076_167572854";
+  private File untarredSegmentFile = new File(INDEX_DIR, SEGMENT_NAME);
+  private File _realtimeSegmentUntarred;
+
+  @BeforeClass
+  @Override
+  public void setUp()
+      throws Exception {
+    TEMP_FILE = File.createTempFile("SegmentCommitterTest", null);
+    // Remove the consumer directory
+    File consumerDirectory = new File(CONSUMER_DIRECTORY);
+    if (consumerDirectory.exists()) {
+      FileUtils.deleteDirectory(consumerDirectory);
+    }
+
+    startZk();
+    startController();
+    addFakeBrokerInstancesToAutoJoinHelixCluster(1, true);
+    addFakeServerInstancesToAutoJoinHelixCluster(1, true);
+
+    ControllerLeaderLocator.create(_helixManager);
+
+    // Start Kafka
+    startKafka();
+    List<File> avroFiles = unpackAvroData(_tempDir);
+
+    setUpRealtimeTable(avroFiles.get(0));
+  }
+
+  @Override
+  protected boolean useLlc() {
+    return true;
+  }
+
+  @Test
+  public void testDefaultCommitter() {
+    IndexLoadingConfig indexLoadingConfig = mock(IndexLoadingConfig.class);
+    ServerMetrics serverMetrics = mock(ServerMetrics.class);
+    doNothing().when(serverMetrics).addMeteredGlobalValue(ServerMeter.LLC_CONTROLLER_RESPONSE_COMMIT_SUCCESS, 1);
+    ServerSegmentCompletionProtocolHandler
+        protocolHandler = new ServerSegmentCompletionProtocolHandler(serverMetrics, getTableName());
+    SegmentCompletionProtocol.Request.Params params = mock(SegmentCompletionProtocol.Request.Params.class);
+    SegmentCompletionProtocol.Response prevResponse = mock(SegmentCompletionProtocol.Response.class);
+    LLRealtimeSegmentDataManager.SegmentBuildDescriptor segmentBuildDescriptor = mock(
+        LLRealtimeSegmentDataManager.SegmentBuildDescriptor.class);
+
+    RealtimeSegmentZKMetadata metadata = _helixResourceManager.getRealtimeSegmentMetadata(getTableName()).get(0);
+
+    when(params.getSegmentName()).thenReturn(metadata.getSegmentName());
+    when(params.getInstanceId()).thenReturn("Server_localhost_0");
+    when(params.getOffset()).thenReturn(END_OFFSET);
+    when(segmentBuildDescriptor.getSegmentTarFilePath()).thenReturn(_realtimeSegmentUntarred + ".tar.gz");
+    when(segmentBuildDescriptor.getBuildTimeMillis()).thenReturn(0L);
+    Map<String, File> metadataPaths = new HashMap();
+    metadataPaths.put(metadata.getSegmentName(), new File(untarredSegmentFile, "metadata.properties"));
+
+    when(segmentBuildDescriptor.getMetadataFiles()).thenReturn(metadataPaths);
+    when(segmentBuildDescriptor.getOffset()).thenReturn(END_OFFSET);
+    when(segmentBuildDescriptor.getSegmentSizeBytes()).thenReturn(0L);
+    when(segmentBuildDescriptor.getWaitTimeMillis()).thenReturn(0L);
+
+    SegmentCompletionManager segmentCompletionManager = _controllerStarter.getSegmentCompletionManager();
```

Comment:
why are you doing this? you call the commit code, after callinug  the segmentConsumed API like I demonstrated with curl command